src/yul.rs

```diff
@@ -1,13 +1,21 @@
 use std::fmt;
 
+#[derive(Hash, Clone, PartialEq, Debug)]
+pub struct SourceLocation {
+    pub start: usize,
+    pub end: usize,
+}
+
 #[derive(Hash, Clone, PartialEq, Debug)]
 pub struct Block {
     pub statements: Vec<Statement>,
+    pub location: Option<SourceLocation>,
```

Comment:
Should the locations be optional or mandatory? Initially I wanted to make them mandatory but there's at least one test that constructs the AST manually (not from a source file) and in that use case you might want to skip them.